sync/src/main/java/tech/pegasys/artemis/sync/SyncManager.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.sync;
+
+import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.compute_start_slot_at_epoch;
+import static tech.pegasys.artemis.statetransition.util.ForkChoiceUtil.on_block;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.primitives.UnsignedLong;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
+import tech.pegasys.artemis.networking.eth2.Eth2Network;
+import tech.pegasys.artemis.networking.eth2.peers.Eth2Peer;
+import tech.pegasys.artemis.networking.eth2.rpc.core.ResponseStream;
+import tech.pegasys.artemis.statetransition.StateTransition;
+import tech.pegasys.artemis.statetransition.StateTransitionException;
+import tech.pegasys.artemis.storage.ChainStorageClient;
+import tech.pegasys.artemis.storage.Store;
+import tech.pegasys.artemis.storage.events.StoreDiskUpdateEvent;
+
+public class SyncManager {
+
+  private final Eth2Network network;
+  private final ChainStorageClient storageClient;
+  private final StateTransition stateTransition;
+  private final EventBus eventBus;
+
+  private Set<Eth2Peer> peerBlacklist;
+
+  public SyncManager(
+      final Eth2Network network, final ChainStorageClient storageClient, final EventBus eventBus) {
+    this.network = network;
+    this.storageClient = storageClient;
+    this.eventBus = eventBus;
+    this.stateTransition = new StateTransition(false);
+    this.peerBlacklist = new HashSet<>();
+  }
+
+  private void sync() {
+    Eth2Peer syncPeer = findSyncPeer().orElseThrow();
+    requestSyncBlocks(
+        syncPeer,
+        ((block) -> {
+          try {
+            Store.Transaction transaction = storageClient.getStore().startTransaction();
+            on_block(transaction, block, stateTransition);
+            transaction.commit();
+            eventBus.post(new StoreDiskUpdateEvent(transaction));
+          } catch (StateTransitionException e) {
+            peerBlacklist.add(syncPeer);
+            sync();
```

Comment:
If the peer is refusing to send us blocks, then sending a status request every time we finish retrieving a set of blocks would leave us in the dark waiting for them. 

How about this: instead of syncing to a single node, send sync requests to all nodes that are ahead of us, skip processing blocks we have seen, keep a peer to latest response time map, and periodically check that map to either blacklist peers or to ask them for their new statuses (for the ones that have too old latest response times). @mbaxter your feedback would be nice as well.